docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,601 @@
+- Python 3 (check the [installation guide](python-tutorial.md#install-a-python-interpreter) if you don’t have it installed)
+- [Python extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-python.python) (For additional details on installing extensions, see [Extension Marketplace](https://code.visualstudio.com/docs/editor/extension-marketplace))
+
+In this section, we will create a folder to be opened as a workspace in VS Code, set up a Python virtual environment, install the project’s dependencies and write the first lines of our FastAPI app.
+
+1.	In your file system, create a project folder for this tutorial, such as `groceries-plugin`.
+2.	Open this new folder in VS Code (**File** > **Open Folder…** or `kb(workbench.action.files.openFolder)`).
+
+
+Now, let’s create a requirements.txt file listing the dependencies we wish to install for the application. The requirements.txt file is a common practice in Python development, used to specify the libraries that your project relies on and their versions. This file helps ensure that anyone working on the project can recreate a similar development environment, making it a convenient component for maintaining consistency across different development environments.
+
+We will want to install FastAPI for creating the app, uvicorn to work as the server, and Redis and type-redis for handling data storage and interacting with a Redis database.
+
+3. Create a new file in VS Code (**File** > **New Text File** or `kb(workbench.action.files.newUntitledFile)`).
+4. Add the following content to it:
+    ```
+    fastapi
+    redis
+    types-redis
+    uvicorn
+    ```
+5.	Save the file (`kb(workbench.action.files.save)`) and name it requirements.txt
+6.	Create a virtual environment by opening the Command Palette (`kb(workbench.action.showCommands)`) and running the **Python: Create Environment** command.
+    > Note: this step may take a couple of minutes to complete.
+7. When asked by the environment type, select **Venv**:
+![Drop down with "Venv" or "Conda" as options for environments that can be created with the Python: Create Environment command.](images/environments/create_environment_dropdown.png)
+
+8. Then select the latest version of Python available on your machine:
+![List of available global environments that can be used to create a virtual environment.](images/fastapi-tutorial/create_environment_interpreters_list.png)
+
+9. Select the requirements.txt file from the drop-down list, so the dependencies are automatically installed, and then select "OK":
+![Check box selected to install dependencies from requirements.txt file](images/fastapi-tutorial/create_environment_select_requirements.png)
+
+The virtual environment will be created, the dependencies will be automatically installed and the environment will be selected for your workspace to be used by the Python extension.
```

Comment:
It would be nice to add steps (if any exist) on how to confirm that the virtual environment has been created successfully and it has been selected in the workbench.

Reply:
Good point. Added a screenshot as well as a note with a workaround in case the Python extension fails to auto select the venv for some reason.